File: 250506/좌우로 움직이는 로봇/robot-moving-from-side-to-side.java
import java.util.Scanner;

public class Main {

    public static final int MAX_N = 1000001;
    public static int[] aMove = new int[MAX_N];
    public static int[] bMove = new int[MAX_N];

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int m = sc.nextInt();
        int aTime = 1;
        int bTime = 1;
        for (int i = 0; i < n; i++) {
            int a = sc.nextInt();
            String a_move = sc.next();
            if (a_move.equals("R")) {
                for (int j = 0; j < a; j++) {
                    aMove[aTime] = aMove[aTime - 1] + 1;
                    aTime++;
                }
            } else {
                for (int j = 0; j < a; j++) {
                    aMove[aTime] = aMove[aTime - 1] - 1;
                    aTime++;
                }
            }
        }
        for (int i = 0; i < m; i++) {
            int b = sc.nextInt();
            String b_move = sc.next();
            if (b_move.equals("R")) {
                for (int j = 0; j < b; j++) {
                    bMove[bTime] = bMove[bTime - 1] + 1;
                    bTime++;
                }
            } else {
                for (int j = 0; j < b; j++) {
                    bMove[bTime] = bMove[bTime - 1] - 1;
                    bTime++;
                }
            }
        }
        if(aTime > bTime) {
            for (int i = bTime; i < aTime; i++) {
                bMove[i] = bMove[bTime - 1];
            }
        } else if(bTime > aTime) {
            for (int i = aTime; i < bTime; i++) {
                aMove[i] = aMove[aTime - 1];
            }
        }
        int meetCount = 0;
        for (int i = 1; i < Math.max(aTime, bTime); i++) {
            if(aMove[i] == bMove[i]) {
                if(i == 1 || (aMove[i - 1] == bMove[i - 1])) {
                    continue;
                }
                meetCount++;
            }
        }
        System.out.println(meetCount);
        sc.close();
    }
}

File: 250506/좌우로 움직이는 로봇/robot-moving-from-side-to-side.cpp
#include <iostream>
#include <cmath>
#define MAX_N 1000001

using namespace std;

int a_pos[MAX_N];
int b_pos[MAX_N];

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    int n, m;
    int a_time = 1, b_time = 1;
    cin >> n >> m;
    for(int i = 0; i < n; i++) {
        int a;
        char a_move;
        cin >> a >> a_move;
        if(a_move == 'R') {
            for(int j = 0; j < a; j++) {
                a_pos[a_time] = a_pos[a_time - 1] + 1;
                a_time++;
            }
        }
        else {
            for(int j = 0; j < a; j++) {
                a_pos[a_time] = a_pos[a_time - 1] - 1;
                a_time++;
            }
        }
    }
    for(int i = 0; i < m; i++) {
        int b;
        char b_move;
        cin >> b >> b_move;
        if(b_move == 'R') {
            for(int j = 0; j < b; j++) {
                b_pos[b_time] = b_pos[b_time - 1] + 1;
                b_time++;
            }
        }
        else {
            for(int j = 0; j < b; j++) {
                b_pos[b_time] = b_pos[b_time - 1] - 1;
                b_time++;
            }
        }
    }

    if(a_time > b_time) {
        for(int i = b_time; i < a_time; i++) {
            b_pos[i] = b_pos[b_time - 1];
        }
    }
    else if(b_time > a_time) {
        for(int i = a_time; i < b_time; i++) {
            a_pos[i] = a_pos[a_time - 1];
        }
    }
    int meetCount = 0;
    for(int i = 1; i < max(a_time, b_time); i++) {
        if(a_pos[i] == b_pos[i]) {
            if(i == 1 || a_pos[i - 1] == b_pos[i - 1]) {
                continue;
            }
            meetCount++;
        }
    }
    cout << meetCount;
    return 0;
}